projects/train/apptainer.def

```diff
@@ -0,0 +1,37 @@
+Bootstrap: docker
+From: python:3.10.12-slim-bullseye
+Stage: build
+
+%files
+# only copy precisely the files we need to keep things
+# light and protect against any large files that might
+# accidentally get created in this repo
+. /opt/amplfi/projects/train
+../../ml4gw /opt/amplfi/ml4gw
+
+%post
+# installing the local package editably via pip
+# works, but none of the dependencies (e.g. ml4gw)
+# are installed editably, so we'll opt for using
+# the full poetry machinery without using a
+# virtual environment. Note that this can potentially
+# cause issues if the poetry install makes changes to
+# the virtualenv package in the base environment, so
+# we'll hard pin it to the value we need.
+# TODO: there's got to be a better solution here
+python -m pip install poetry
+cd /opt/amplfi/projects/train
+poetry export -o requirements.txt --without-hashes \
```

Comment:
Why not simply
```
python -m pip install --no-deps -r <(poetry export --without-hashes)
pip install --no-deps .
```

Reply:
Yup, think that one liner would work too let me try it out